### PR TITLE
TRAN-60: Add Intl support on frontend

### PR DESCRIPTION
**Pull request by KarolinaKopacz**
**Add Intl support on frontend**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.export = {
+    rules: {
+        'max-len': ["error", { "code": 120 }]
+    }
+}
```
Comment (sniedzielski) [inline .eslintrc.js]: Can you add empty space at the end of file?

File: src/_tests_/components/header/HeaderMenu.test.tsx [hunk 2]
```diff
@@ -4,17 +4,24 @@ import { render, screen } from '@testing-library/react';
...
+import { IntlProvider } from 'react-intl';
+import en from 'translations/en.json';
 
 describe('Unit tests for the HeaderMenu component.', () => {
   const DASHBOARD_TEST_ID = 'dashboard-link';
   const REPORTS_TEST_ID = 'reports-link';
   const MANUAL_UPLOAD_TEST_ID = 'manual-upload-link';
   const EXCEL_UPLOAD_TEST_ID = 'excel-upload-link';
 
+  const locale = 'en'
```
Comment (sniedzielski) [inline src/_tests_/components/header/HeaderMenu.test.tsx]: can you moved those const into 'utils/consts'?

File: src/_tests_/components/header/HeaderMenu.test.tsx [hunk 3]
```diff
@@ -4,17 +4,24 @@ import { render, screen } from '@testing-library/react';
...
+  const messages = { 'en': en }
+
   test('Dashboard link should to have specific href attribute.', () => {
     render(
       <MemoryRouter initialEntries={[{ pathname: '/', search: '/' }]}>
-        <HeaderMenu />
+        <IntlProvider locale={locale} defaultLocale='en' messages={messages[locale]}>
```
Comment (sniedzielski) [inline src/_tests_/components/header/HeaderMenu.test.tsx]: Please use const n 'defaultLocale='en''

File: src/_tests_/components/header/HeaderMenu.test.tsx [hunk 4]
```diff
@@ -25,7 +32,9 @@ describe('Unit tests for the HeaderMenu component.', () => {
   test('Reports link should to have specific href attribute.', () => {
     render(
       <MemoryRouter initialEntries={[{ pathname: '/', search: '/' }]}>
-        <HeaderMenu />
+        <IntlProvider locale={locale} defaultLocale='en' messages={messages[locale]}>
```
Comment (sniedzielski) [inline src/_tests_/components/header/HeaderMenu.test.tsx]: Please use const n 'defaultLocale='en''

File: src/_tests_/components/header/HeaderMenu.test.tsx [hunk 5]
```diff
@@ -47,7 +58,9 @@ describe('Unit tests for the HeaderMenu component.', () => {
   test('Excel Upload link should to have specific href attribute.', () => {
     render(
       <MemoryRouter initialEntries={[{ pathname: '/', search: '/' }]}>
-        <HeaderMenu />
+        <IntlProvider locale={locale} defaultLocale='en' messages={messages[locale]}>
```
Comment (sniedzielski) [inline src/_tests_/components/header/HeaderMenu.test.tsx]: Please use const n 'defaultLocale='en''

File: src/_tests_/pages/signIn/SignInPage.test.tsx [hunk 6]
```diff
@@ -6,16 +6,26 @@ import { MemoryRouter } from 'react-router-dom';
...
 import SignInPage from 'pages/signIn/SignInPage';
 import store from 'stores/store';
+import { IntlProvider } from 'react-intl';
+import en from 'translations/en.json';
+import { CookiesProvider } from 'react-cookie';
 
 describe('Unit tests for the SignInPage component.', () => {
   const SIGN_IN_PAGE_TEST_ID = 'sign-in-page';
 
+  const locale = 'en'
```
Comment (sniedzielski) [inline src/_tests_/pages/signIn/SignInPage.test.tsx]: can you moved those const into 'utils/consts'?

File: src/_tests_/components/logout/Logout.test.tsx [hunk 7]
```diff
@@ -4,14 +4,21 @@ import { render, screen } from '@testing-library/react';
 import { MemoryRouter } from 'react-router-dom';
 
 import Logout from 'components/logout/Logout';
+import { IntlProvider } from 'react-intl';
+import en from 'translations/en.json';
 
 describe('Unit test for the Logout component.', () => {
   const LOGOUT_TEST_ID = 'logout';
 
+  const locale = 'en'
```
Comment (sniedzielski) [inline src/_tests_/components/logout/Logout.test.tsx]: can you moved those const into 'utils/consts'?

File: src/_tests_/shared/dialog/Dialog.test.tsx [hunk 8]
```diff
@@ -15,6 +17,9 @@ describe('Unit test for the Dialog component.', () => {
   const mockOnAdd = jest.fn();
   const mockOnCancel = jest.fn();
 
+  const locale = 'en'
```
Comment (sniedzielski) [inline src/_tests_/shared/dialog/Dialog.test.tsx]: can you moved those const into 'utils/consts'?

File: src/components/forms/signInForm.tsx [hunk 9]
```diff
@@ -24,10 +25,13 @@ type AuthAction = {
   type: string,
   error?: unknown
 };
-
+// @ts-ignore
```
Comment (sniedzielski) [inline src/components/forms/signInForm.tsx]: Is this necessary?

File: src/pages/signIn/SignInPage.tsx [hunk 10]
```diff
@@ -1,13 +1,37 @@
...
+import { useCookies } from 'react-cookie';
+import Select from 'react-select';
+
 import SignInForm from 'components/forms/signInForm';
 
 export default function SignInPage() {
+  const [langCookies, setLangCookie] = useCookies(['language']);
+
+  const options = [
+    { value: 'en', label: 'English' },
```
Comment (sniedzielski) [inline src/pages/signIn/SignInPage.tsx:12]: maybe we can take advantage of having const from 'utils/consts' if you fix consts related to this values.

File: src/_tests_/shared/dialog/Dialog.test.tsx [hunk 11]
```diff
@@ -15,6 +17,9 @@ describe('Unit test for the Dialog component.', () => {
   const mockOnAdd = jest.fn();
   const mockOnCancel = jest.fn();
 
+  const locale = 'en';
```
Comment (sniedzielski) [inline src/_tests_/shared/dialog/Dialog.test.tsx]: Can we move those constants into 'utils/consts'?

File: src/_tests_/pages/signIn/SignInPage.test.tsx [hunk 12]
```diff
@@ -1,21 +1,31 @@
...
 import { MemoryRouter } from 'react-router-dom';
 
 import SignInPage from 'pages/signIn/SignInPage';
 import store from 'stores/store';
+import en from 'translations/en.json';
 
 describe('Unit tests for the SignInPage component.', () => {
   const SIGN_IN_PAGE_TEST_ID = 'sign-in-page';
 
+  const locale = 'en';
```
Comment (sniedzielski) [inline src/_tests_/pages/signIn/SignInPage.test.tsx]: Can we move those constants into 'utils/consts'?

File: src/_tests_/pages/signIn/SignInPage.test.tsx [hunk 13]
```diff
@@ -1,21 +1,31 @@
...
+  const messages = { en };
+
   test('Should match snapshot.', () => {
     render(
       <Provider store={store}>
-        <MemoryRouter initialEntries={[{ pathname: '/sign_in', search: '/' }]}>
-          <SignInPage />
-        </MemoryRouter>
+        <CookiesProvider>
+          <MemoryRouter initialEntries={[{ pathname: '/sign_in', search: '/' }]}>
```
Comment (sniedzielski) [inline src/_tests_/pages/signIn/SignInPage.test.tsx]: can we change this: pathname: '/sign_in' -> 'Paths.sign_in' etc?

File: src/App.tsx [hunk 14]
```diff
@@ -1,14 +1,34 @@
...
+
 function App() {
+  const [cookies] = useCookies();
+
   return (
-    <div className="bg-transit-grey w-screen h-screen">
-      <RoutesConfig />
-    </div>
+    <IntlProvider
+      locale={cookies.language.value}
```
Comment (sniedzielski) [inline src/App.tsx]: please use default language when there is no such cookie with saved selected language etc

File: src/App.tsx [hunk 15]
```diff
@@ -1,14 +1,34 @@
...
+      defaultLocale="en"
```
Comment (sniedzielski) [inline src/App.tsx:23]: please use const maybe const EN = 'en' and defined this in utils/const.ts

File: src/App.tsx (hunk 14 above)
Comment (sniedzielski) [reply src/App.tsx]: there is error 'Uncaught TypeError: cookies.language is undefined'

**Review by sniedzielski — CHANGES_REQUESTED** [review]
@KarolinaKopacz Please refer to my comments. Changes in tests are 'nice to have'. Please take a look into 'App.tsx' line 24.

File: src/pages/signIn/SignInPage.tsx (hunk 10 above)
Comment (KarolinaKopacz) [reply src/pages/signIn/SignInPage.tsx:12]: This is use only in one place

**Review by sniedzielski — APPROVED** [review]
Now looks good and it seems that error with 'cookies' undefinied is resolved (I tested this locally). To be checked on 'dev' instace after deployment.

**Pull request merged**